suitest.js

```diff
@@ -49,36 +49,38 @@ const {NETWORK_PROP, NETWORK_METHOD} = require('./lib/constants/networkRequest')
 const HAD_NO_ERROR = require('./lib/constants/hadNoError');
 
 // Network
-const {webSocketsFactory} = require('./lib/api/webSockets');
-const {unusedExpressionWatchersFactory} = require('./lib/utils/unusedExpressionWatchers');
+const webSocketsFactory = require('./lib/api/webSockets');
+const unusedExpressionWatchersFactory = require('./lib/utils/unusedExpressionWatchers');
 const ipcClient = require('./lib/testLauncher/ipc/client');
 const ipcServer = require('./lib/testLauncher/ipc/server');
 
 // Contexts
 const Context = require('./lib/utils/Context');
 const AuthContext = require('./lib/utils/AuthContext');
 const {configFactory} = require('./config');
-
+const {setUpRaven} = require('./lib/utils/sentry/Raven');
 const {warnLauncherAndLibHasDiffVersions} = require('./lib/utils/packageMetadataHelper');
 const {createLogger} = require('./lib/utils/logger');
 
 // Publicly available API goes here
-class SUITEST_API {
-	constructor(exitOnError, instanceDependencies) {
-		// default instance dependencies
+class Suitest {
+	constructor({exitOnError}) {
+		// instance dependencies
 		this.authContext = new AuthContext();
 		this.appContext = new Context();
 		this.pairedDeviceContext = new Context();
 		this.testContext = new Context();
 		this.configuration = configFactory();
 		this.config = this.configuration.config;
-		// override default by provided
-		Object.assign(this, instanceDependencies);
+		this.configuration.configurableFields.map(fieldName => {
+			this[`set${fieldName[0].toUpperCase()}${fieldName.slice(1)}`] = (val) => this.configuration.override({[fieldName]: val});
```

Comment:
why not create single method that will accept fieldName that need to change and value? Also, need to update .d.ts file.

Reply:
I added it d.ts file, Andrey sugested it, previous configure was confusing as we have other methods to set configuration, so instead of one several was introduced

Reply:
Ok, but you defined only 4 methods for this, but configurableFileds contains 
'logLevel', 'logDir', 'timestamp', 'configFile', 'continueOnFatalError', 'disallowCrashReports', 'defaultTimeout'.
I whould define single function for all fields, or separate function for each field explicitly for reduce complexety, currently I need jump throught codebase for understanding what going on,.

Reply:
Updated configurableFields, did this way to have a list of fields that user can configure during execution